packages/syft/src/syft/client/client.py

```diff
@@ -415,45 +414,69 @@ def apply_to_gateway(self, client: Self) -> None:
 
     @property
     def data_subject_registry(self) -> Optional[APIModule]:
-        if self.api is not None and hasattr(self.api.services, "data_subject"):
+        if self.api is not None and self.api.has_service("data_subject"):
```

Comment:
Added a helper function `has_service(name)` to make this check a bit cleaner.

Aditionally, By looking at the code `self.api is not None` doesn't seem to be None. Because invoking `self.api` always invokes `_fetch_api()`. Let me know if this is redundant, I'll remove it.

Reply:
❤️ , I think we can remove it

Reply:
We don't always fetch, we cache it in self._api where self.api acts a property pointing to self._api